cmd/otelcol/main.go

```diff
@@ -169,66 +155,119 @@ func checkRuntimeParams() {
+	ballastSize := getKeyValue(os.Args[1:], "--mem-ballast-size-mib")
 	if ballastSize != "" {
-		config = os.Getenv(ballastEnvVarName)
-		if config != "" {
-			log.Fatalf("Both %v and '--config' were specified, but only one is allowed", ballastEnvVarName)
+		if os.Getenv(ballastEnvVarName) != "" {
+			log.Fatalf("Both %v and '--mem-ballast-size-mib' were specified, but only one is allowed", ballastEnvVarName)
 		}
 		os.Setenv(ballastEnvVarName, ballastSize)
 	}
 	setMemoryBallast(memTotalSizeMiB)
 	setMemoryLimit(memTotalSizeMiB)
 }
 
-// Validate and set the configuration
-func setConfig() {
-	// Check if the config is specified via the env var.
-	config := os.Getenv(configEnvVarName)
-	// If not attempt to use a default config; supports Docker and local
-	if config == "" {
-		_, err := os.Stat(defaultDockerSAPMConfig)
-		if err == nil {
-			config = defaultDockerSAPMConfig
+// Validate then set config file path environment variable to specified config file path flag.
+// Validate then set the config file path flag to the specified config file path environment variable if not specified.
+// Calls setValidDefaultConfigPath to validate and set the config file path flag, config file path environment variable
+// to the default config file path if the config file path flag, config file path environment variable and config YAML
+// environment variable are not specified.
+func setValidConfigPath() {
+	configPathFlag := getKeyValue(os.Args[1:], "--config")
+	configPathVar := os.Getenv(configEnvVarName)
+	configYaml := os.Getenv(configYamlEnvVarName)
+
+	if configPathFlag == "" && configPathVar == "" {
+		if configYaml == "" {
+			setValidDefaultConfigPath()
+		} else {
+			log.Printf("Using environment variable %s for configuration", configYamlEnvVarName)
 		}
-		_, err = os.Stat(defaultLocalSAPMConfig)
-		if err == nil {
-			config = defaultLocalSAPMConfig
+		return
+	}
+
+	if configPathFlag != "" {
+		validateConfigPath(configPathFlag)
+
+		if configPathVar != "" && configPathVar != configPathFlag {
+			log.Printf("Both %v and '--config' were specified. Overriding %q environment variable value with %q for this session", configEnvVarName, configPathVar, configPathFlag)
 		}
-		if config == "" {
-			log.Fatalf("Unable to find the default configuration file, ensure %s environment variable is set properly", configEnvVarName)
+		os.Setenv(configEnvVarName, configPathFlag)
+
+		if configYaml != "" {
+			log.Printf("Both %s and '--config' were specified. Using '--config' value %s for this session", configYamlEnvVarName, configPathFlag)
 		}
-	} else {
-		// Check if file exists.
-		_, err := os.Stat(config)
-		if err != nil {
-			log.Fatalf("Unable to find the configuration file (%s) ensure %s environment variable is set properly", config, configEnvVarName)
+
+		log.Printf("Set config to %v", configPathFlag)
+		return
+	}
+
+	if configPathVar != "" {
+		validateConfigPath(configPathVar)
+
+		if !contains(os.Args[1:], "--config") {
```

Comment:
This should always be the case since you're returning where `configPathFlag != ""` right?

Reply:
No, this is when configPathFlag == "" and configPathVar != ""

Reply:
`configPathFlag == ""` means that `!contains(os.Args[1:], "--config")` will always be true, so why check?